x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobTriggerMethod.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector.syncjob;
+
+import java.util.Locale;
+
+public enum ConnectorSyncJobTriggerMethod {
+    ON_DEMAND,
+    SCHEDULED;
+
+    public static ConnectorSyncJobTriggerMethod fromString(String triggerMethodString) {
+        for (ConnectorSyncJobTriggerMethod triggerMethod : ConnectorSyncJobTriggerMethod.values()) {
+            if (triggerMethod.name().equalsIgnoreCase(triggerMethodString)) {
+                return triggerMethod;
+            }
+        }
+
+        return null;
```

Comment:
Should we fail fast if we are provided illegal `triggerMethodString`?  Should we throw `IllegalArgumentException` if we are unable to parse the string?

Reply:
That makes sense! Adjusted with https://github.com/elastic/elasticsearch/pull/102714/commits/3536d74ea02921008d016b029ac176e7e54d978c.